src/commandHandlers/support.ts

```diff
@@ -20,8 +32,77 @@ export const command = async (
     );
 };
 
-export const description = 'Get Support information';
+const parseSearchCommand = (text: string) =>
+  text.match(/^search\s+([\s\S]+)$/i);
+
+const searchIssuesCommand = async (question: string, embed: MessageEmbed) => {
+  const words = question.split(/\W+/gi).filter((word) => word);
+  const keywords = removeStopwords(words);
+  const issues = await fetchIssues(keywords.join(' '));
+
+  embed.setColor(COLORS.github).setTitle('Support repository issues');
+
+  if (!issues.length) {
+    return embed.setDescription(
+      "we couldn't find any issue, raise a question on the [support repo](https://github.com/EddieJaoudeCommunity/support/issues/new/choose)."
+    );
+  }
+
+  embed.setDescription(
+    "if you didn't find what you need, raise a question on the [support repo](https://github.com/EddieJaoudeCommunity/support/issues/new/choose)."
+  );
+
+  issues.forEach((issue: Issue) => {
+    const description = stripMarkdown(issue.body).trim().replace(/\n+/g, '\n');
+    const briefDescription = truncate(description);
+    embed.addField(
+      `${issue.title} #${issue.number}`,
+      `Author: [${issue.user.login}](${issue.user.html_url}) | Status: ${issue.state} | Comments: ${issue.comments}\n[${briefDescription}](${issue.html_url})`
+    );
+  });
+
+  return embed;
+};
+
+const fetchIssues = async (q: string) => {
+  const request = {
+    ...issueRequestConfig,
+    url: '/issues',
+    params: {
+      per_page: 3,
+      q: `${q} is:issue repo:EddieJaoudeCommunity/support`,
```

Comment:
_⚠️ just an idea for the future that I'd very much love opinions on 😃. This PR is already awesome and adding value! ⚠️_

Having the "repo:EddieJaoudeCommunity/support" and "is:issue" filters could be extracted out of this function, and added through **composition**. Something like this:
```ts
const fetchIssues = async (getAdditionalFiltersFn: (string) => string) => (q: string) =>  {
  const request = {
    params: {
      q: getAdditionalFiltersFn(q)
    }
  }
  //...
}

const fromRepoFilter = (repo: string) => (q: string) => `repo:${repo} ${q}`;
const isIssueFilter = (q: string) => `is:issue ${q}`;
const hasLabelFilter = (label: string) => (q: string) => `label:${label} ${q}`;

const hasDocumentation = hasLabelFilter("documentation");
const supportRepo = fromRepoFilter("repo:EddieJaoudeCommunity/support");
const fetchEddieJaoudeCommunitySupportIssues = fetchIssues(
  q => isIssueFilter(hasDocumentation(supportRepo(q)))
);

// Usage
fetchEddieJaoudeCommunitySupportIssues("review my project");
```
The idea is that the `fetchIssues` function can be more generic and used throughout the codebase. Therefore, I think this idea only makes sense once this usage becomes a necessity! Generalizing now would make the code harder to reason about, but it's just an idea for the future 😃 

Let me know any of your thoughts on this.

Reply:
I love this idea 👍 😁, we already have some duplication in this PR in [src/commandHandlers/issue.ts#L8](https://github.com/EddieJaoudeCommunity/EddieBot/blob/ac2e28f170e0dbf602d3b5186b2ffa29d209211f/src/commandHandlers/issue.ts#L8-L25) and [src/commandHandlers/support.ts#L67](https://github.com/EddieJaoudeCommunity/EddieBot/blob/ac2e28f170e0dbf602d3b5186b2ffa29d209211f/src/commandHandlers/support.ts#L67-L80) that composition will defiantly fix it.  
we could also use [Fluent APIs](https://www.linkedin.com/pulse/chain-your-javascript-methods-like-boss-fluent-apis-anthony-gore/) 
<details>
  <summary>Example</summary>

```typescript
class Github {
  private request: AxiosRequestConfig;
  private filters: Set<string>;

  constructor() {
    this.filters = new Set();
    this.request = {
      ...issueRequestConfig,
      baseURL: `https://api.github.com`,
      url: '',
      params: {},
    };
  }

  search(): Github {
    this.request.url += '/search';
    return this;
  }

  issues(): Github {
    this.request.url += '/issues';
    this.filters.add('is:issue');
    return this;
  }

  inSupportRepo(): Github {
    this.filters.add('repo:EddieJaoudeCommunity/support');
    return this;
  }

  limit(perPage: number): Github {
    this.request.params.per_page = perPage;
    return this;
  }

  async get(q: string): Promise<Issue[]> {
    this.filters.add(q);
    this.request.params.q = [...this.filters].join(' ');
    try {
      return (await axios(this.request)).data.items;
    } catch (error) {
      log.error('Something went wrong when fetching the issues: ', error);
      return [];
    }
  }
}

// Usage
new Github().search().issues().inSupportRepo().limit(3).get("review my project")
```

</details>

But composition is way better since it's way more generic, someone should open a PR for it 😄

Reply:
Having a Fluent APIs is a great idea 💯.

 I think we can open a new issue for refactoring purposes, and with a link to your comment. Using a class to encapsulate the responsibility of interacting with the GitHub Web API would be better since we are separating the two concerns. The files you mentioned can be simplified and abstracted of that logic 😃.

Reply:
The issue #418 was created for this 😃